server/src-lib/Hasura/RQL/DDL/Metadata.hs

```diff
@@ -352,18 +381,23 @@ fetchMetadata = do
   functions <- map (uncurry QualifiedObject) <$>
     Q.catchE defaultTxErrorHandler fetchFunctions
 
-  -- fetch all custom resolvers
-  schemas <- DRS.fetchRemoteSchemas
-
+  -- fetch all remote schemas
+  remoteSchemas <- DRS.fetchRemoteSchemas
+    -- fetch all remote schema perms
+  remoteSchemaPerms <- DRS.fetchRemoteSchemaPerms
```

Comment:
fetch remote schemas along with their permission meta in `SQL`? Since `RemoteSchemaPermMeta` has `FromJSON` instance we can write `SQL` in such a way that it emits an array of json objects instead of merging them in Haskell code

Reply:
I considered this but it was getting ugly to write a query which conforms to RemoteSchemaMeta mostly because it's not a simple JOIN alone, you also need to aggregate on the permissions so that one remote schema -> one row. This seemed much simpler. Do you see any problem with this approach?

Reply:
You can make use of `json_agg` with `group by`. It is much safer to do this in SQL in one shot :) Also fast too.

Reply:
I agree in general that is true. But here, I think, it is much safer to make 2 simple queries vs 1 complex query. Also, there is no need to be concerned about performance here.

This is the SQL btw, seems like we have just transferred the complexity from the code to the SQL which seems to me to be a worse bargain:

```
SELECT rs.name, rs.definition::json, rs.comment, rsp.perms::json
FROM hdb_catalog.remote_schemas rs
LEFT JOIN (
     SELECT remote_schema, json_agg(json_build_object ('role', role, 'definition', definition)) as perms
     FROM hdb_catalog.remote_schema_permissions rsp
     GROUP BY remote_schema
      ) rsp
ON rs.name = rsp.remote_schema
 ```

Reply:
If you still prefer this, I can make this change.

Reply:
If Postgres can able to give me correct `JSON`s then I'm happy to parse them as Haskell values and use them directly. Preparing the SQL query makes it fast. We cannot "prepare" Haskell code ;)